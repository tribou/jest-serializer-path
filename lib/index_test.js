'use strict'

const Serializer = require('./')

describe('serializer', () => {

  it('replaces process.cwd with <PROJECT_ROOT>', () => {

    const sut = process.cwd()
    expect(sut).toMatchSnapshot()

  })


  it('preserves path directories after the <PROJECT_ROOT>', () => {

    const sut = `${process.cwd()}/src/somewhere`
    expect(sut).toMatchSnapshot()

  })


  it('replaces process.cwd with <PROJECT_ROOT> in Object properties', () => {

    const sut = {
      myPath: `${process.cwd()}/src`,
    }
    expect(sut).toMatchSnapshot()

  })

  it('replaces process.cwd with <PROJECT_ROOT> in array', () => {

    const sut = [`${process.cwd()}/src`]
    expect(sut).toMatchSnapshot()

  })

  it('supports trailing slashes in the path', () => {

    const sut = `${process.cwd()}/path/with/trailing/slash/`
    expect(sut).toMatchSnapshot()

  })


  it('replaces every instance of process.cwd in the same string', () => {

    const sut = {
      PATH: `${process.cwd()}/path:${process.cwd()}/another/path`,
      script: `const myPath = ${process.cwd()}/path;
      const mySecondPath = ${process.cwd()}/another/path;`,
    }
    expect(sut).toMatchSnapshot()

  })

})


describe('serializer.test()', () => {

  it('returns true when val is a string and contains process.cwd', () => {

    const val = `${process.cwd()}/a/path`
    const result = Serializer.test(val)
    expect(result).toEqual(true)

  })

  it('returns false when val is a string and does not contain process.cwd', () => {

    const val = '/a/path'
    const result = Serializer.test(val)
    expect(result).toEqual(false)

  })

  it('returns true when val is an object with a property that contains process.cwd', () => {

    const val = {
      property: `${process.cwd()}/a/path`,
      property2: '/no/dirname',
    }
    const result = Serializer.test(val)
    expect(result).toEqual(true)

  })

  it('returns false when val is an object with a property that does not contain process.cwd', () => {

    const val = {
      property: '/a/path/',
      property2: '/no/dirname',
    }
    const result = Serializer.test(val)
    expect(result).toEqual(false)

  })

  it('returns true when val is an array with a property that contains process.cwd', () => {

    const val = [`${process.cwd()}/a/path`, '/no/dirname']
    const result = Serializer.test(val)
    expect(result).toEqual(true)

  })

  it('returns false when val is an array with a property that does not contain process.cwd', () => {

    const val = ['/a/path/', '/no/dirname']
    const result = Serializer.test(val)
    expect(result).toEqual(false)

  })

})


describe('serializer.print()', () => {

  it('replaces process.cwd with <PROJECT_ROOT> when val is a string', () => {

    const ser = jest.fn()
    const val = `${process.cwd()}/a/path`
    const expected = '<PROJECT_ROOT>/a/path'
    Serializer.print(val, ser)
    expect(ser.mock.calls[0][0]).toEqual(expected)

  })

  it('replaces process.cwd with <PROJECT_ROOT> in string property when val is an object', () => {

    const ser = jest.fn()
    const val = {
      property: `${process.cwd()}/a/path`,
      property2: '/no/dirname',
    }
    const expected = {
      property: '<PROJECT_ROOT>/a/path',
      property2: '/no/dirname',
    }
    Serializer.print(val, ser)
    expect(ser.mock.calls[0][0]).toEqual(expected)

  })

  it('replaces process.cwd with <PROJECT_ROOT> in string property when val is an array', () => {

    const ser = jest.fn()
    const val = [`${process.cwd()}/a/path`, '/no/dirname']
    const expected = ['<PROJECT_ROOT>/a/path', '/no/dirname']
    Serializer.print(val, ser)
    expect(ser.mock.calls[0][0]).toEqual(expected)

  })

})
