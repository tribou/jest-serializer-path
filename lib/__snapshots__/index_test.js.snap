// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer preserves path directories after the <PROJECT_ROOT> 1`] = `"<PROJECT_ROOT>/src/somewhere"`;

exports[`serializer replaces every instance of process.cwd in the same string 1`] = `
Object {
  "PATH": "<PROJECT_ROOT>/path:<PROJECT_ROOT>/another/path",
  "script": "const myPath = <PROJECT_ROOT>/path;
      const mySecondPath = <PROJECT_ROOT>/another/path;",
}
`;

exports[`serializer replaces process.cwd with <PROJECT_ROOT> 1`] = `"<PROJECT_ROOT>"`;

exports[`serializer replaces process.cwd with <PROJECT_ROOT> in Object properties 1`] = `
Object {
  "myPath": "<PROJECT_ROOT>/src",
}
`;

exports[`serializer replaces process.cwd with <PROJECT_ROOT> in array 1`] = `
Array [
  "<PROJECT_ROOT>/src",
]
`;

exports[`serializer replaces process.cwd with <PROJECT_ROOT> in an Error 1`] = `[Error: some error in <PROJECT_ROOT>/a/path]`;

exports[`serializer supports trailing slashes in the path 1`] = `"<PROJECT_ROOT>/path/with/trailing/slash/"`;
